js/robotClasses.js

```diff
@@ -0,0 +1,78 @@
+"use strict";
+
+let $ = require("jquery"),
+    selection = require("./robotSelection");
```

Comment:
Require statements need to be a one-way street. It won't work to have modules requiring each other. You have this requiring 'robotSelection`, but that file is requiring this one.